bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/discovery/RokuDiscoveryService.java

```diff
@@ -0,0 +1,267 @@
+            if (response.contains(ROKU_SSDP_MATCH)) {
+                parseResponseCreateThing(response);
+            }
+        } while (!socketTimeout);
+    }
+
+    /**
+     * Process the response from the Roku into a DiscoveryResult.
+     *
+     */
+    private void parseResponseCreateThing(String response) {
+        DiscoveryResult result;
+
+        String label = "Roku";
+        String uuid = null;
+        String host = null;
+        int port = -1;
+
+        Scanner scanner = new Scanner(response);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] pair = line.split(":", 2);
+            if (pair.length != 2) {
+                continue;
+            }
+
+            String key = pair[0].toLowerCase();
+            String value = pair[1].trim();
+            logger.debug("key: {} value: {}.", key, value);
+            switch (key) {
+                case "location":
+                    host = value;
+                    Matcher matchIp = IP_HOST_PATTERN.matcher(value);
+                    if (matchIp.find()) {
+                        host = matchIp.group(1);
+                        port = Integer.parseInt(matchIp.group(2));
+                    }
+                    break;
+                case "usn":
+                    Matcher matchUid = USN_PATTERN.matcher(value);
+                    if (matchUid.find()) {
+                        uuid = matchUid.group(2);
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+        scanner.close();
+
+        if (host == null || port == -1 || uuid == null) {
+            logger.debug("Bad Format from Roku, received data was: {}", response);
+            return;
+        } else {
+            logger.debug("Found Roku, uuid: {} host: {}", uuid, host);
+        }
+
+        uuid = uuid.replace(":", "").toLowerCase();
+
+        ThingUID thingUid = new ThingUID(THING_TYPE_ROKU_PLAYER, uuid);
+
+        // Try to query the device using discovered host and port to get extended device info
+        try {
+            RokuCommunicator communicator = new RokuCommunicator(httpClient, host, port);
+            DeviceInfo device = communicator.getDeviceInfo();
+            label = device.getModelName() + " " + device.getModelNumber();
+            if (device.isTv())
+                thingUid = new ThingUID(THING_TYPE_ROKU_TV, uuid);
+        } catch (RokuHttpException e) {
+            logger.debug("Unable to retrieve Roku device-info. Exception: {}", e.getMessage(), e);
+        }
+
+        result = DiscoveryResultBuilder.create(thingUid).withLabel(label).withRepresentationProperty(PROPERTY_UUID)
+                .withProperty(PROPERTY_UUID, uuid).withProperty(PROPERTY_HOST_NAME, host)
+                .withProperty(PROPERTY_PORT, port).build();
+        this.thingDiscovered(result);
```

Comment:
At this point there is enough information to create the thing. This call being caught here only enhances the result. I would like to leave it as-is just in case the call to retrieve the extended info stops working due to a device firmware update or similar issue.